backend/objstore/health_objstore.go

```diff
@@ -52,18 +53,38 @@ func (h *HealthCheck) HealthCheck(ctx context.Context) servicehealthcheck.Health
 	}
 
 	// Try download
-	obj, err := h.Client.GetObjectWithContext(
+	obj, err := h.Client.GetObject(
 		ctx,
 		cfg.HealthCheckBucketName,
 		cfg.HealthCheckObjectName,
-		minio.GetObjectOptions{},
+		minio.GetObjectOptions{
+			// if version is given, get the specific version
+			// otherwise version should be empty
+			VersionID: info.VersionID,
+		},
 	)
 	if err != nil {
 		h.state.SetErrorState(fmt.Errorf("failed to get object: %v", err))
 		return h.state.GetState()
 	}
 	defer obj.Close()
 
+	// in case the bucket is versioned, delete versions to not leak versions
+	if info.VersionID != "" {
```

Comment:
This only downloads the last version, correct?

Reply:
It only deletes objects when they are versioned.

Reply:
Yes but only the `latest` version, because the above code would only return the `latest` version, correct? That's what my question was about.